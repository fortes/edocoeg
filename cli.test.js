jest.mock('./geo-rename', () => {
  return {tagFiles: jest.fn(() => Promise.resolve(true))};
});

const {tagFiles} = require('./geo-rename');

const {main} = require('./cli');

beforeEach(() => {
  console._originalError = console.error;
  console._originalLog = console.log;
  console.error = jest.fn(() => {});
  console.log = jest.fn(() => {});
});

afterEach(() => {
  console.error = console._originalError;
  console.log = console._originalLog;
});

describe('argument handling', () => {
  test('usage info if no arguments', async () => {
    await main([]);
    expect(console.log).toHaveBeenCalled();
    expect(console.log.mock.calls[0][0]).toMatchSnapshot();
  });
});
