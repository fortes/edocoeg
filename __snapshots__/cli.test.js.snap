// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`argument handling usage info if no arguments 1`] = `
"geo-rename [OPTIONS] FILE
Find location information from GPS and write to EXIF

  --cache-folder <path>
  -f, --force
  -h, --help
  --offline
  -q, --quiet
  -s, --skip-backup
  -v, --verbose
  -V, --version
"
`;
